### ISSUE TITLE
Provide built distributions at pypi.org

### ISSUE THREAD

**Issue by jrueb** [xrootd/xrootd#1833]
Would it be possible to provide built distributions at pypi.org so that the user doesn't have to compile xrootd themselves?
This would make the installation much easier. For example I was trying to install xrootd through pip and got an error telling me to install openssl and uuid development packages. It would be extremely convenient if I did not have to figure out how to install though with my current setup.

**Comment by simonmichal** [comment]
@jrueb : we currently don't have a python/pip packager on the team, but of course we would be open to a PR!

**Comment by amadio** [comment]
Fixed by #2025 (can now generate binary artifacts, will close this when we upload to PyPI).

**Comment by amadio** [comment]
I still need to figure out how to upload binary wheels for the various Linux distributions, but at least for macOS I could upload a wheel that should work with the latest version of Homebrew:

```
% pip3 install --user xrootd==5.6.0
Collecting xrootd==5.6.0
  Downloading xrootd-5.6.0-cp311-cp311-macosx_11_0_x86_64.whl (5.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.0/5.0 MB 9.0 MB/s eta 0:00:00
Installing collected packages: xrootd
Successfully installed xrootd-5.6.0

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: python3.11 -m pip install --upgrade pip
% python3
Python 3.11.4 (main, Jun 20 2023, 16:52:35) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyxrootd
>>> dir(pyxrootd.client)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'pyxrootd' has no attribute 'client'
>>> from pyxrootd import client
>>> client.
client.CopyProcess(       client.EnvGetString_cpp(  client.File(              client.XrdVersion_cpp(   
client.EnvGetDefault_cpp( client.EnvPutInt_cpp(     client.FileSystem(       
client.EnvGetInt_cpp(     client.EnvPutString_cpp(  client.URL(              
>>> client.XrdVersion_cpp()
'5.6.0'
```

I will close this ticket for now, upload the remaining binary wheels, and for next release I will make it part of the release process which will run in the CI environment and upload automatically.
